src/main/java/seedu/nuscents/data/TransactionList.java

```diff
@@ -79,8 +80,37 @@ public void filterTransaction(TransactionCategory category) {
         } else {
             Ui.showFilterNotFoundMessage(category);
         }
+    }
 
+    public void setBudget(float budget) {
+        this.budget = budget;
+        Ui.showBudgetSet(this);
     }
 
-}
+    public float getBudget() {
+        return this.budget;
+    }
 
+    public float getTotalExpense() {
+        float totalExpense = 0;
+        ArrayList<Transaction> transactions = getTransactions();
+        for (Transaction transaction : transactions) {
+            int index = transactions.indexOf(transaction) + 1;
+            if (transaction instanceof Expense) {
+                totalExpense += transaction.getAmount();
+            }
+        }
+        return totalExpense;
+    }
+    public boolean isWithinBudget() {
+        boolean isWithinBudget;
+        float budget = getBudget();
+        float expense = getTotalExpense();
+        if (expense > budget) {
+            isWithinBudget = false;
+        } else {
+            isWithinBudget = true;
+        }
+        return isWithinBudget;
+    }
+}
```

Comment:
Perhaps you could add a newline between the 2 methods

Reply:
Oh ok, will edit it to make it more readable!